util/weekly_build/ShellSetup.sh

```diff
@@ -0,0 +1,101 @@
+echo Run ID: ${1?"First arg: Unique run ID"}
+echo Run directory: ${2?"Second arg: base directory for build"}
+echo Platform name: ${3?"Third arg: platform name ('hera', 'hercules', etc.)"}
+
+RUNID=$1
+RUNDIR=$2
+PLATFORM=$3
+
+if [ ${RUNDIR::1} != "/" ]; then
+  echo "FATAL ERROR: Directory should be an absolute path!"
+  exit 1
+fi
+
+PACKAGES_TO_TEST=${PACKAGES_TO_TEST:-"libpng libaec jasper scotch w3emc g2 g2c"}
+
+function alert_cmd {
+  echo "Your run failed in $1. This is a placeholder alerting function. 'alert_cmd' should be defined for each system."
+}
+
+function spack_install_exe {
+  spack $* | tee -a log.install 2>&1
+}
+
+case $PLATFORM in
+  hercules)
+    COMPILERS=${COMPILERS:-"intel gcc"}
+    BUILD_CACHE_DIR=${BUILD_CACHE_DIR:-/work/noaa/epic/role-epic/spack-stack/hercules/build_cache}
+    ;;
+  orion)
+    COMPILERS=${COMPILERS:-"intel gcc"}
+    BUILD_CACHE_DIR=${BUILD_CACHE_DIR:-/work/noaa/epic/role-epic/spack-stack/orion/build_cache}
+    ;;
+  discover16)
+    COMPILERS=${COMPILERS:-"intel gcc"}
+    BUILD_CACHE_DIR=${BUILD_CACHE_DIR:-/gpfsm/dswdev/jcsda/spack-stack/scu16/build_cache}
+    ;;
+  discover17)
+    COMPILERS=${COMPILERS:-"intel gcc"}
+    BUILD_CACHE_DIR=${BUILD_CACHE_DIR:-/gpfsm/dswdev/jcsda/spack-stack/scu17/build_cache}
+    ;;
+  derecho)
+    COMPILERS=${COMPILERS:-"intel gcc"}
+    BUILD_CACHE_DIR=${BUILD_CACHE_DIR:-/glade/work/epicufsrt/contrib/spack-stack/derecho/build_cache}
+    ;;
+  acorn)
+    module load gcc/11.2.0 python/3.11.7
```

Comment:
I think we should move those to separate preamble scripts. Not sure where to store those best, but there needs to be a structure like sites/tierN/platform/compiler since these preamble scripts are platform and compiler specific (at least for cray).

Reply:
If it's okay I'd lean toward waiting to see how complicated the config for each site gets, then if it's more than a couple lines per site break it out into separate scripts.

Reply:
See readthedocs for Narwhal - different blocks for each compiler, both of them long

Reply:
Done. I didn't break them into tier1/tier2 just to keep the logic simple.

Reply:
I think that makes sense, we'll probably only do this for tier1 anyway